src/openvic-simulation/dataloader/Dataloader.cpp

```diff
@@ -948,69 +948,41 @@ bool Dataloader::load_defines(GameManager& game_manager) {
 		Logger::error("Failed to load diplomatic actions!");
 		ret = false;
 	}
+	if(!game_manager.get_script_manager().get_condition_manager().setup_conditions(game_manager)) {
+		Logger::error("Failed to set up conditions!");
+		ret = false;
+	}
 
-	parse_scripts(game_manager);
+	ret &= parse_scripts(game_manager);
 
 	free_cache();
 
 	return ret;
 }
 
+#define PARSE_SCRIPTS(name, parse) \
+	if (!parse) { Logger::error("Failed to parse ", name, " scripts!"); ret = false; } \
+	else Logger::info("Successfully parsed ", name, " scripts!");
```

Comment:
```suggestion
#define PARSE_SCRIPTS(name, parse) \
	if (!parse.parse_scripts(game_manager)) { Logger::error("Failed to parse ", name, " scripts!"); ret = false; } \
	else Logger::info("Successfully parsed ", name, " scripts!");
```
You could move the `parse_scripts` call to the macro and just pass the manager to it

Reply:
didnt think of it, nice